### PR TITLE
[2810] Align the naming of Slot API parameters

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2810

### 🎯 Goal

Align the naming of Slot API parameters.

### 🛠 Implementation details

The parameters were aligned according to the rules below:

For 1 slot (like wrapping something): content
For 3 horizontal slots: leadingContent / centerContent / trailingContent
For 3 vertical slots: headerContent / centerContent / footerContent
For 5 slots: leadingContent / headerContent / centerContent / footerContent/ trailingContent
For 2 vertical slots: headerContent / centerContent OR centerContent / footerContent

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
~~- [ ] Changelog is updated with client-facing changes~~
~~- [ ] New code is covered by unit tests~~
~~- [ ] Comparison screenshots added for visual changes~~
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt [hunk 1]
```diff
@@ -65,8 +65,8 @@ public fun ChannelItem(
     leadingContent: @Composable RowScope.(ChannelItemState) -> Unit = {
         DefaultChannelAvatar(channelItem = it, currentUser = currentUser)
     },
-    detailsContent: @Composable RowScope.(ChannelItemState) -> Unit = {
-        DefaultChannelDetails(
+    centerContent: @Composable RowScope.(ChannelItemState) -> Unit = {
+        DefaultChannelCenterContent(
```
Comment (filbabic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt:69]: Maybe `DefaultChannelItemX...`

So

`DefaultChannelItemLeadingContent`˛
`DefaultChannelItemCenterContent`˛
`DefaultChannelItemTrailingContent`˛

We have the DefaultChannelAvatar here which should be leading content

**Review by filbabic — COMMENTED** [review]
Left one small comment!

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt:69]: Yes, agree, there is no issue with having lengthy names. Better for consistency.

**Review by filbabic — APPROVED** [review]
Looks good!

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

Can we also please update the changelog?

**Pull request merged**